sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala

```diff
@@ -31,7 +31,7 @@ import org.apache.spark.sql.internal.SQLConf
  * @since 2.0.0
  */
 @Stable
-class RuntimeConfig private[sql](sqlConf: SQLConf = new SQLConf) {
+class RuntimeConfig private[sql](val sqlConf: SQLConf = new SQLConf) {
```

Comment:
Why do we need this change and the one is connect client compatibility above ?

Reply:
This is probably the smallest change building a new StateStoreConf via copying SQL conf & applying offset metadata configs.

```
        val clonedRuntimeConf = new RuntimeConfig(session.sessionState.conf.clone())
        OffsetSeqMetadata.setSessionConf(metadata, clonedRuntimeConf)
        StateStoreConf(clonedRuntimeConf.sqlConf)
```

OffsetSeqMetadata.setSessionConf requires RuntimeConfig, the constructor of StateStoreConf requires SQLConf.

Reply:
Yes, the above change is about connect compatibility. I don't think Spark connect will leverage this anyway.

Reply:
hmm, @cloud-fan is this change safe?